pandas/core/internals.py

```diff
@@ -1807,16 +1807,20 @@ def _try_operate(self, values):
         return values.view('i8')
 
     def _try_coerce_args(self, values, other):
-        """ provide coercion to our input arguments
-            we are going to compare vs i8, so coerce to integer
-            values is always ndarra like, other may not be """
+        """ Coerce values and other to dtype 'i8'. NaN and NaT convert to
+            the smallest i8, and will correctly round-trip to NaT if converted
+            back in _try_coerce_result. values is always ndarray-like, other
+            may not be """
         values = values.view('i8')
+
```

Comment:
its not that we *need* to infer, my point is that a very similar code block exists in several places in the code base. needs to be consolidated to avoid repeating is again. Pls look thru and see where it is duplicated (e.g. in internals.py as well)